aavf-spec.md

```diff
@@ -0,0 +1,216 @@
+| Ref     |           g l K k s |    gga ctc AAA aaa tcc | K is the reference amino acid                             |
+| 1       |           g l - k s |    gga ctc --- aaa tcc | K amino acid is deleted w.r.t. to the reference sequence  |
+| 2       |           g l KKk s | gga ctc AAAAAA aaa tcc | K amino acid is inserted w.r.t. to the reference sequence |
+
+Representing these as AAVF records would be done as follows:
+
+  1. A single amino acid deletion of K at position 3 becomes REF=LK, ALT=L
+  2. A single amino acid insertion of K after position 3 becomes REF=K, ALT=KK
+
+Note: that the positions must be sorted in increasing order:
+
+``` 
+#CHROM    GENE    POS     REF    ALT     FILTER  ALT_FREQ  COVERAGE INFO
+my_chrom  a       2       LK     L       PASS    0.5       1000     RC=ctcaaa;AC=ctc
+my_chrom  a       3       K      KK      PASS    0.5       1000     RC=aaa;AC=aaaaaa;ACF=0.5
+```
+
+### 3.4 Decoding AAVF entries for Insertions and Deletions
+
+#### 3.4.1 Insertion AAVF record
+
+Supposed I receive the following AAVF record:
+
+``` 
+#CHROM    GENE    POS     REF    ALT     FILTER  ALT_FREQ  COVERAGE INFO
+my_chrom  a       3       K      KK      PASS    0.5       1000     RC=aaa;AC=aaaaaa;ACF=0.5
+```
+
+This is an insertion since the reference amino acid K is being replaced by K [the reference amino acid] plus one insertion amino acid K in such a way that a gap is opened in the reference. Again there are only two alleles so I have the two following segrating haplotypes.
+
+| Example | Amino Acid Sequence | Nucleotide Sequence    | Alteration                                                |
+|---------|--------------------:|-----------------------:|-----------------------------------------------------------|
+| Ref     |           g l K k s | gga ctc AAA --- aaa tcc | K is the reference amino acid                             |
+| 1       |           g l KKk s | gga ctc AAA AAA aaa tcc | K amino acid is inserted w.r.t. to the reference sequence |
+
+#### 3.4.2 Deletion AAVF record
+
+Supposed I receive the following AAVF record:
+
+```
+#CHROM    GENE    POS     REF    ALT     FILTER  ALT_FREQ  COVERAGE INFO
+my_chrom  a       2       LK     L       PASS    0.5       1000     RC=ctcaaa;AC=ctc
+```
+
+This is a deletion of one reference amino acid since the reference allele LK is being replaced by just the L [the reference amino acid]. Again there are only two alleles so I have the two following seggregating haplotypes.
+
+| Example | Amino Acid Sequence | Nucleotide Sequence    | Alteration                                                |
+|---------|--------------------:|-----------------------:|-----------------------------------------------------------|
+| Ref     |           g l K k s |    gga ctc AAA aaa tcc | K is the reference amino acid                             |
+| 1       |           g l - k s |    gga ctc --- aaa tcc | K amino acid is deleted w.r.t. to the reference sequence  |
```

Comment:
Just having some thoughts on the way to annotate indels using left normalization. This approach implicitly uses two-codon linkage information. In the examples, where the left codon is not changed it is clear how to use the format but how can it be used (and specified) when the left codon is also changing. 

For instance, in this latter example, "l" aminoacid could be coded by 6 codons (TTG,TTA,CT{A,T,C,G}) and some of the linked deletions may occur in one or another. should then the INFO field be populated by as many as codon combination exists. 

What if the left aminoacid also changes. Using the same example:
Ref     |           g l K k s |    gga ctc AAA aaa tcc | K is the reference amino acid                             
1         |           g l - k s |    gga ctc --- aaa tcc | K amino acid is deleted w.r.t. to the reference sequence  
2         |           g l - k s |    gga ctg --- aaa tcc | K amino acid is deleted w.r.t. to the reference sequence  
3         |           g F - k s |    gga ttc --- aaa tcc | K amino acid is deleted w.r.t. to the reference sequence  
4        |           g F - k s |    gga ttc AAA aaa tcc | K amino acid is deleted w.r.t. to the reference sequence 

Would the format then be:

RC=ctcaaa,AC=ctc,ctg,ttc,ttcAAA,ACF=0.25,0.25,0.25,0.25

or just:
RC=ctcaaa,AC=ctc,ctg,ttc,ACF=0.25,0.25,0.25